### ISSUE TITLE
Add support for extending units to `languageOptions`

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#9070]
### What is the problem you're trying to solve?

I'd like to add my own units, e.g.

```css
a { top: 10--rpx; }
```

And have rules like [`declaration-property-value-no-unknown`](https://stylelint.io/user-guide/rules/declaration-property-value-no-unknown) not report them as a problem.

Ref: https://github.com/stylelint/stylelint/issues/9037#issuecomment-3891721635

### What solution would you like to see?

A new `units` property within `syntax` on the `languageOptions` to extend things like `length`, `time` etc.

It seems CSSTree [allows configuring units](https://github.com/csstree/csstree/blob/56afb6dd761149099cd3cdfb0a38e15e8cc0a71a/lib/__tests/lexer.js#L129-L148) during the creation of the lexer:

```js
const customLexer = createLexer({
	generic: true,
	cssWideKeywords: ['wide'],
	units: {
		length: ['aa', 'bb'],
	},
	types: {
		foo: '<number> | <length>',
	},
	properties: {
		test: '<foo>+',
	},
});

it('custom syntax should not affect base syntax', () => {
	assert.strictEqual(lexer.validate(), null);
	assert.strictEqual(lexer.matchProperty('test', '1 2aa 3').matched, null);
	assert.strictEqual(lexer.matchProperty('test', 'wide').matched, null);
	assert.notStrictEqual(lexer.matchProperty('color', 'red').matched, null);
});
```

Is this something we can use?

**Comment by tofrankie** [comment]
Looking forward to it... 👀

**Comment by ybiquitous** [comment]
The new property `languageOptions.syntax.units` would be like this, right? Sounds good to me. 👍🏼 

```json
{
  "languageOptions": {
    "syntax": {
      "units": {
        "length": ["foo", "bar"]
      }
    }
  }
}
```


Ref https://github.com/stylelint/stylelint/blob/49ee611975d8648a31f4fddd6f62b63b2a74f9dc/docs/user-guide/configure.md#languageoptions
